### ISSUE TITLE
Find a way to make import dynamic to not have useless warning from SDK like facebook who say wrong inited

### ISSUE THREAD

**Issue by riderx** [Cap-go/capacitor-social-login#28]

**Comment by reslear** [comment]
I agree with @piotr-cz, yes it will add work on unified types, but it will be a much more competent solution

**Comment by riderx** [comment]
I think we could use a hook for this case @WcaleNieWolny ?

**Comment by ToriChanIntegration** [comment]
Thanks for raising this one. I agree this warning noise is annoying in real projects.

Concrete next step proposal:
- Lazy-load provider SDKs only when that provider is used
- Keep current API stable
- Add an opt-in flag first so we can test behavior safely

If you still see this on the latest version, could you drop one exact warning line and the provider it comes from? That will help us pin the first provider to migrate to dynamic loading.
